nipype/pipeline/engine/workflows.py

```diff
@@ -0,0 +1,1096 @@
+                        subnodename = subnodefullname.replace('.', '_')
+                        for _ in self._graph.get_edge_data(node,
+                                                           subnode)['connect']:
+                            dotlist.append('%s -> %s;' % (nodename,
+                                                          subnodename))
+                        logger.debug('connection: ' + dotlist[-1])
+        # add between workflow connections
+        for u, v, d in self._graph.edges_iter(data=True):
+            uname = '.'.join(hierarchy + [u.fullname])
+            vname = '.'.join(hierarchy + [v.fullname])
+            for src, dest in d['connect']:
+                uname1 = uname
+                vname1 = vname
+                if isinstance(src, tuple):
+                    srcname = src[0]
+                else:
+                    srcname = src
+                if '.' in srcname:
+                    uname1 += '.' + '.'.join(srcname.split('.')[:-1])
+                if '.' in dest and '@' not in dest:
+                    if not isinstance(v, Workflow):
+                        if 'datasink' not in \
+                           str(v._interface.__class__).lower():
+                            vname1 += '.' + '.'.join(dest.split('.')[:-1])
+                    else:
+                        vname1 += '.' + '.'.join(dest.split('.')[:-1])
+                if uname1.split('.')[:-1] != vname1.split('.')[:-1]:
+                    dotlist.append('%s -> %s;' % (uname1.replace('.', '_'),
+                                                  vname1.replace('.', '_')))
+                    logger.debug('cross connection: ' + dotlist[-1])
+        return ('\n' + prefix).join(dotlist)
+
+    def _plain_connect(self, *args, **kwargs):
+        super(ConditionalWorkflow, self).connect(*args, **kwargs)
+
+
+class ConditionalWorkflow(Workflow):
+    """
+    Implements a kind of workflow that can be by-passed if the input of
+    `donotrun` of the condition node is `True`.
+    """
+
+    def __init__(self, name, base_dir=None):
+        """Create a workflow object.
+        Parameters
+        ----------
+        name : alphanumeric string
+            unique identifier for the workflow
+        base_dir : string, optional
+            path to workflow storage
+        """
+
+        from nipype.interfaces.utility import IdentityInterface
+        super(ConditionalWorkflow, self).__init__(name, base_dir)
+        self._condition = Node(IdentityInterface(fields=['donotrun']),
+                               name='checknode')
+        self.add_nodes([self._condition])
+
+    def _check_conditional_nodes(self):
+        from nipype.interfaces.utility import IdentityInterface
+
+        def _checkdefined(val):
+            from nipype.interfaces.base import isdefined
+            if isdefined(val):
+                return bool(val)
+            return False
+
+        if checknode is None:
```

Comment:
Yes, since https://github.com/oesteban/nipype/commit/6f3a3aa2cf8c95ba08af3a95b95430419c63ef91 I'm doing a bit of refactoring, now that I have the implementation clear.